src/transport/RendezvousSession.cpp

```diff
@@ -64,13 +65,83 @@ RendezvousSession::~RendezvousSession()
     mDelegate = nullptr;
 }
 
-CHIP_ERROR RendezvousSession::SendMessage(System::PacketBuffer * buffer)
+CHIP_ERROR RendezvousSession::SendMessage(System::PacketBuffer * msgBuf)
 {
-    // Rendezvous does not use a MessageHeader yet, but the Transport::Base API expects one, so
-    // let build an empty one for now.
+    CHIP_ERROR err = mPairingInProgress ? SendPairingMessage(msgBuf) : SendSecureMessage(msgBuf);
+    SuccessOrExit(err);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        mPairingInProgress ? OnPairingError(err) : OnRendezvousError(err);
+    }
+    return err;
```

Comment:
```C
SuccessOrExit(err);
mPairingInProgress ? OnPairingError(err) : OnRendezvousError(err);

exit:
    return err;
```

Thoughts on rewriting as such?

Reply:
I can't really, because ```mPairingInProgress ? OnPairingError(err) : OnRendezvousError(err);``` should only be executed when there is an error and ```SuccessOrExit(err)```will goto ```exit``` in this case.